libraries/core_libs/consensus/include/transaction/transaction_manager.hpp

```diff
@@ -101,10 +100,10 @@ class TransactionManager : public std::enable_shared_from_this<TransactionManage
    *
    * @note transaction might be already processed -> they are not processed and inserted again
    * @param tx transaction to be processed
-   * @param status transaction status
-   * @return true if successfully inserted unseen transactions
+   * @param insert_non_proposable insert non proposable transactions
+   * @return transaction status
    */
-  bool insertValidatedTransaction(std::shared_ptr<Transaction> &&tx, const TransactionStatus status);
+  TransactionStatus insertValidatedTransaction(std::shared_ptr<Transaction> &&tx, bool insert_non_proposable = true);
```

Comment:
The use case is that when gossiped transactions are received we insert them and have true. We don't insert them with false only when rpc does it. This is the only two places this is used. I think I left the default option on true because there is a lot of tests that use it with the true option.